src/pharmpy/model/execution_steps.py

```diff
@@ -272,6 +264,32 @@ def _canonicalize_and_check_method(method: str) -> str:
             )
         return method
 
+    @staticmethod
+    def _canonicalize_derivatives(derivatives: tuple) -> tuple:
+        if isinstance(derivatives, tuple):
```

Comment:
Hard to follow because of too deep nesting. Check for the problematic case and raise and have no `else`

Reply:
Restructured for clarity